### PR TITLE
bump client-go and k8s api versions

### PR DESCRIPTION
**Pull request by tomkennedy513**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,23 +4,23 @@ go 1.14
...
+	k8s.io/api v0.19.7
+	k8s.io/apimachinery v0.19.7
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	knative.dev/pkg v0.0.0-20200702222342-ea4d6e985ba0
+	knative.dev/pkg v0.0.0-20210331065221-952fdd90dbb0
 	sigs.k8s.io/yaml v1.2.0
 )
 
-replace k8s.io/client-go => k8s.io/client-go v0.17.5
+replace k8s.io/client-go => k8s.io/client-go v0.19.7
```
Comment (tylerphelan) [inline go.mod:26]: this seems to imply we don't need the replace anymore: https://github.com/pivotal/kpack/pull/681/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R54

**Review by tylerphelan — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -4,23 +4,23 @@ go 1.14
...
-	github.com/evanphx/json-patch v4.5.0+incompatible
+	github.com/evanphx/json-patch v4.9.0+incompatible
 	github.com/ghodss/yaml v1.0.0
-	github.com/google/go-cmp v0.5.1
-	github.com/google/go-containerregistry v0.1.1
+	github.com/google/go-cmp v0.5.5
+	github.com/google/go-containerregistry v0.4.1
 	github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
-	github.com/pivotal/kpack v0.1.5-0.20210107191603-fa0a973272f5
+	github.com/pivotal/kpack v0.2.3-0.20210430201452-ed7fe5056666
```
Comment (matthewmcnew) [inline go.mod:12]: Nice

File: go.mod (hunk 1 above)
Comment (tomkennedy513) [reply go.mod:26]: seeing this failure after removing it and running go mod tidy 
```
go: finding module for package k8s.io/api/auditregistration/v1alpha1
github.com/pivotal/build-service-cli/pkg/testhelpers imports
        k8s.io/client-go/kubernetes/fake imports
        k8s.io/api/auditregistration/v1alpha1: module k8s.io/api@latest found (v0.21.0), but does not contain package k8s.io/api/auditregistration/v1alpha1
```

File: pkg/clusterstack/factory.go [hunk 3]
```diff
@@ -38,8 +39,8 @@ type Factory struct {
 	Repository string
 }
 
-func NewFactory(cs k8s.ClientSet, ch *commands.CommandHelper, rup registry.UtilProvider, tlsCfg registry.TLSConfig) (*Factory, error) {
-	repo, err := k8s.DefaultConfigHelper(cs).GetCanonicalRepository()
+func NewFactory(ctx context.Context, cs k8s.ClientSet, ch *commands.CommandHelper, rup registry.UtilProvider, tlsCfg registry.TLSConfig) (*Factory, error) {
+	repo, err := k8s.DefaultConfigHelper(cs).GetCanonicalRepository(ctx)
```
Comment (matthewmcnew) [inline pkg/clusterstack/factory.go:43]: Sigh. It feels like this in unexpected (A constructor making a network call). The Context makes it more obvious.

File: go.mod (hunk 1 above)
Comment (tomkennedy513) [reply go.mod]: was able to remove it by bumping to the 0.19.7 release of client-go

**Review by matthewmcnew — APPROVED** [review]
Looks great!

**Pull request merged**